src/nasher.nim

```diff
@@ -115,11 +115,13 @@ when isMainModule:
             unpack(opts, target)
           else:
             var updatedNss: seq[string]
+            var exitCode = QuitSuccess
             if convert(opts, target, updatedNss) and
-               compile(opts, target, updatedNss) and
+               compile(opts, target, updatedNss, exitCode) and
                pack(opts, target) and
                install(opts, target):
                  launch(opts, target)
+            quit(exitCode)
```

Comment:
I've added a gate around the exit condition so that it is only called if the exit code is non-zero.

This seems like the easiest solution, however it will mean that if target 1 has a compilation error you will never get to `target2` or `target3`.  I _think_ that's fine, we are effectively using the fail fast principle here so that people can quickly find and fix errors.

If you want to allow people to continue to process additional targets even if the first one fails that sounds more complex.  You also start getting into situations where if `target2` relies on `target1` the first failure is going to break everything anyway.

I guess if you want to try and add more complex functionality it would be good to try and understand the overall intent and what the edge cases you are trying to cover are.